Refactor, update deps, and add tests.

gcoakes opened the pull request:
I was having some issues with the transformer on the latest parcel version and decided to update a few things. It was failing on `config.shouldReload()`. I removed that, and proceeded to get a bit carried away... I touched most parts of the repo, so I understand if you don't want to take all the changes. However, I think the tests and `shouldReload` changes at the very least should be considered.

Please let me know if you want any or all of this. For now, I'm happy to just use my fork for my personal projects, so I understand if you reject this on the basis of it being too big of a change.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/orlov-vo/parcel-transformer-svelte/24)
<!-- Reviewable:end -->

gcoakes committed:
Refactor, update deps, and add tests.

gcoakes committed:
Upgrade deps.

orlov-vo commented:
Thanks for your work! But, I can't accept / make good review of the PR with all-in-one changes at this time. Can you split your changes for separate PRs with fixes, refactoring, tests and updating deps?

gcoakes committed:
Major version change.

Add typescript preprocessing by default.
Switch to ESM for source and build to common with a ESM shim.

gcoakes commented:
I opened a PR (parcel-bundler/parcel#7508) to the parcel repo itself so svelte can be enabled by default in parcel. I also updated my personal version of this package to do typescript preprocessing out of the box. I don't intend to pursue this PR anymore.
